### PR TITLE
Overhaul of CLE and addition of anisotropic fracture

### PR DESCRIPTION
**Pull request by jameskermode**
I'll leave this as a draft until the corresponding PR to JuLIP is merged (coming up next)

### CODE REVIEW

**Comment by jameskermode** [comment]
Corresponding JuLIP PR is JuliaMolSim/JuLIP.jl#144

**Comment by jameskermode** [comment]
These changes are quite extensive so happy to discuss anything before merging.

The old `elastic_moduli()` was out by a factor of -2. I found the factor of 2, which came from only setting the upper triangle of the deformation matrix and not the lower triangle as well as required for a symmetric strain. However, I couldn't find the minus sign nor understand how the +h followed by inv(-2h) deformations were supposed to work, so I've rewritten in a simpler way that also returns the Atoms to their original state.

Anisotropic fracture CLE is ported from matscipy, and appears to give the correct decay in the residual forces of $r^{-2}$, at least for a single lattice. Haven't thought about how to extend to a multi-lattice yet.

This PR also enables GitHub CI, but since the tests now require `dev` version of JuLIP it's not passing for me. Not sure if there's a proper way to specify that in `Project.toml`, or if I should just add `Pkg.develop("JuLIP")` to the build step.

File: src/CLE.jl [hunk 1]
```diff
@@ -87,16 +100,57 @@ Methods:
...
+voigt_stress(calc::AbstractCalculator, at::AbstractAtoms) = voigt_stress(stress(calc, at))
+voigt_stress(at::AbstractAtoms) = voigt_stress(stress(at))
+
+"""
+Convert from 3×3 strain matrix ϵ to Voigt 6-vector
+"""
+voigt_strain(ϵ::AbstractMatrix) = (@assert size(ϵ) == (3,3);
+                                   reshape(ϵ + ϵ' - Diagonal(ϵ), :)[voigtinds])
+
+# FIXME maybe there's a better name for 3x3 stress tensor than `cauchy_stress`?
```
Comment (jameskermode) [inline src/CLE.jl:138]: Up for debate: is there a better name for the 3x3 matrix forms of stress and strain, as opposed to the Voigt 6-vectors?

File: test/runtests.jl [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
 
 tests = [
     "test_greensfunctions.jl",
     "test_gfcorrectors3D.jl",
     "test_dislocations.jl",
-    "test_forceconstants.jl"
-    # "test_cauchyborn1.jl",
+    "test_elasticity.jl",
+    "test_fracture.jl",
+   # "test_forceconstants.jl"
```
Comment (jameskermode) [inline test/runtests.jl:21]: I fixed the missing files, force constant tests were still failing...

**Comment by jameskermode** [comment]
CI now passing: https://github.com/jameskermode/JuLIPMaterials.jl/actions/runs/1150066628

**Comment by cortner** [comment]
sorry I'd forgotten about this. I'll merge it now and test it.

**Pull request merged**